src/components/Receiving/ItemDetails.js

```diff
@@ -156,58 +236,88 @@ class ItemDetails extends Component {
     const poLineIdsList = Object.keys(lineItems);
     const poLineId = poLineIdsList[currentLine];
     const poLineNumber = get(lineItems, [poLineId, 0, 'poLineNumber'], '');
     const title = get(lineItems, [poLineId, 0, 'title'], '');
+    const isReviewScreen = currentLine >= poLineIdsList.length;
 
     return (
-      <Modal
-        label={
-          (currentLine >= poLineIdsList.length)
-            ? <FormattedMessage id="ui-orders.receiving.reviewDetails" />
-            : <FormattedMessage id="ui-orders.receiving.modalPaneTitle" values={{ poLineNumber, title }} />
-        }
-        footer={
-          <ItemDetailsFooter
-            close={close}
-            onClickNext={this.onClickNext}
-            onClickPrevious={this.onClickPrevious}
-            poLineIdsListLenght={poLineIdsList.length}
-            currentLine={currentLine}
-            checkedItemsListLength={checkedItems.length}
-          />
-        }
-        open
-      >
-        {(this.state.currentLine > poLineIdsList.length)
-          ? <div>Receive Details</div>
-          : (
-            <MultiColumnList
-              contentData={lineItems[poLineId]}
-              formatter={this.getResultsFormatter()}
-              visibleColumns={['isChecked', 'barcode', 'format', 'location', 'itemStatus']}
-              columnMapping={{
-                isChecked: <Checkbox type="checkbox" checked={allChecked[poLineId]} onChange={() => this.toggleAll(poLineId)} />,
-                barcode: <FormattedMessage id="ui-orders.receiving.barcode" />,
-                format: <FormattedMessage id="ui-orders.receiving.format" />,
-                location: <FormattedMessage id="ui-orders.receiving.location" />,
-                itemStatus: <FormattedMessage id="ui-orders.receiving.itemStatus" />,
-              }}
-              columnWidths={{
-                isChecked: '5%',
-                barcode: '25%',
-                format: '20%',
-                location: '35%',
-                itemStatus: '15%',
-              }}
+      <div data-test-item-details>
+        <Modal
+          label={
+            isReviewScreen
+              ? <FormattedMessage id="ui-orders.receiving.reviewDetails" />
+              : <FormattedMessage id="ui-orders.receiving.modalPaneTitle" values={{ poLineNumber, title }} />
+          }
+          footer={
+            <ItemDetailsFooter
+              checkedItemsListLength={checkedItems.length}
+              close={close}
+              currentLine={currentLine}
+              isReceiveButtonDisabled={this.isReceiveButtonDisabled()}
+              onClickNext={this.onClickNext}
+              onClickPrevious={this.onClickPrevious}
+              onClickReceive={this.onClickReceive}
+              poLineIdsListLenght={poLineIdsList.length}
             />
-          )
-        }
-      </Modal>
+          }
+          open
+        >
+          {isReviewScreen
+            ? <ReviewDetails
```

Comment:
I see n inconsistency here. you've separated MCL of Review in a separate component, but Piece Details business logic and render code is here. It's kind of mess with separation of concerns principle. Maybe it makes sense to refactor this.
If there is no big difference between lists it might be a good thing to provide corresponding attributes to it and leave in render just `<MultiColumnList ...props />`

Reply:
I've created two separated components for Line Details and Review Details